### PR TITLE
Fix uninitialized variables warnings

### PR DESCRIPTION
**Pull request by vladmos**
For-loop variables are not guaranteed to be initialized after the loop, but are initialized inside it.

### CODE REVIEW

File: warn/warn_control_flow_test.go [hunk 1]
```diff
@@ -641,4 +641,20 @@ def foo(y):
...
+    print(x.attr)
+
+  print(x)
+  print(x.attr)
+`,
+		[]string{
+			":6: Variable \"x\" may not have been initialized.",
+			":7: Variable \"x\" may not have been initialized.",
+		},
+		scopeEverywhere)
```
Comment (laurentlb) [inline warn/warn_control_flow_test.go]: Add another test:

```
for x in y:
    a = x
    print(a)
print(a)
```

File: warn/warn_control_flow.go [hunk 2]
```diff
@@ -401,7 +401,16 @@ func findUninitializedVariables(stmts []build.Expr, previouslyInitialized map[st
 			// Traverse but ignore the result. Even if something is defined inside a for-loop, the loop
```
Comment (laurentlb) [inline warn/warn_control_flow.go]: Comment is slightly confusing:
```diff
- the loop will be empty or not
+ the collection will be empty or not
```

Comment (vladmos) [reply warn/warn_control_flow.go]: Done.

File: warn/warn_control_flow_test.go (hunk 1 above)
Comment (vladmos) [reply warn/warn_control_flow_test.go]: Done.

**Comment by allevato** [comment]
The 0.22 release should include this fix, correct? But we're still seeing the incorrect lint warnings using this version: https://travis-ci.org/bazelbuild/rules_swift/jobs/493391714

Unfortunately this has made our CI status go red. Could someone look into it and cut a new release if necessary?

**Comment by vladmos** [comment]
This looks like a different bug, I'll send a fix asap.

Has it started to fail only with 0.22?

**Comment by allevato** [comment]
It looks like it. We have a daily cron and the buildifier test started failing today, after the 0.22 release was cut.

It looks like there may be a couple new "legitimate" failures in the mix as well (we have to use the legacy provider syntax in a couple places because `instrumented_files` doesn't have a declared provider counterpart yet), so I'll have to see if I can selectively disable those.

**Comment by vladmos** [comment]
FYI if you don't need a warning you can exclude it by providing `--warnings=-rule-impl-return`.

**Pull request merged**